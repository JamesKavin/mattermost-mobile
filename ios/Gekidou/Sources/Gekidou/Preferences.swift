//
//  Preferences.swift
//  Gekidou
//
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
//
import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
