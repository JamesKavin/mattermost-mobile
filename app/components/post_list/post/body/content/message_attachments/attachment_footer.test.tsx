// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@constants/preferences';
import {render} from '@test/intl-test-helper';

import AttachmentFooter from './attachment_footer';

describe('AttachmentFooter', () => {
    const baseProps = {
        text: 'This is the footer!',
        icon: 'https://images.com/image.png',
        theme: Preferences.THEMES.denim,
    };

    test('it matches snapshot when footer text is provided', () => {
        const props = {
            ...baseProps,
            icon: undefined,
        };

        const wrapper = render(<AttachmentFooter {...props}/>);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('it matches snapshot when both footer and footer_icon are provided', () => {
        const wrapper = render(<AttachmentFooter {...baseProps}/>);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
