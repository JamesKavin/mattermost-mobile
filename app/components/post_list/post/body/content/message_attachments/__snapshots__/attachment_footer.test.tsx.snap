// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentFooter it matches snapshot when both footer and footer_icon are provided 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 5,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 12,
          "marginRight": 5,
          "marginTop": 1,
          "width": 12,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "uri": "https://images.com/image.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "rgba(63,67,80,0.5)",
        "fontSize": 11,
      }
    }
  >
    This is the footer!
  </Text>
</View>
`;

exports[`AttachmentFooter it matches snapshot when footer text is provided 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "marginTop": 5,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "rgba(63,67,80,0.5)",
        "fontSize": 11,
      }
    }
  >
    This is the footer!
  </Text>
</View>
`;
